content/docs/pipelines/caching.md

```diff
@@ -0,0 +1,96 @@
++++
+title = "Caching"
+description = "Getting started with Kubeflow Pipelines step caching"
+weight = 50
++++
+
+Starting from Kubeflow Pipelines 0.4, Kubeflow Pipelines supports step caching capabilities in both standalone deployment and GCP hosted deployment.
+
+## Before you start
+
+This guide tells you the basic concepts of Kubeflow Pipelines step caching and how to use it. This guide assumes that you already have Kubeflow Pipelines installed or want to use standalone or GCP hosted deployment options in the [Kubeflow Pipelines deployment 
+guide](/docs/pipelines/installation/) to deploy Kubeflow Pipelines.
+
+## What is step caching?
+
+Kubeflow Pipelines caching provides step-level output caching on [Kubeflow Pipelines SDK](/docs/pipelines/sdk) only. A step output will be taken from cache if this step template and input does not change. The change includes step name, input artifacts, parameters and underlying logic change. Step output passed in a GCS bucket has not been included in this cache support yet. A cached entry is managed by Kubeflow Pipelines database and will never expire by default. To control caching behavior and change cache entry staleness, please see sections below. With step caching, the output of a certain step can be re-used and we can save the efforts of re-executing previous steps.  
```

Comment:
It might be better to have each sentence on its own line so that it's easier to comment.

>Kubeflow Pipelines caching provides step-level output caching on [Kubeflow Pipelines SDK](/docs/pipelines/sdk) only.

I'm not sure I fully understand what's conveyed by this sentence.
Do this mean that "Execution caching is enabled by default for all pipelines submitted through the KFP backend and UX. The exception is pipelines authored using TFX SDK which has its own caching mechanism."

>The change includes step name, input artifacts, parameters and underlying logic change.

The cache key is based on the component (base image, command-line, code), arguments passed to the component (values or artifacts) and any additional customizations. If the component is exactly the same and the arguments are exactly the same as in some previous execution, then the task can be skipped and the outputs of the old execution can be used.

>Step output passed in a GCS bucket has not been included in this cache support yet.

?

>A cached entry is managed by Kubeflow Pipelines database and will never expire by default.

This is a bit vague.

>To control caching behavior and change cache entry staleness, please see sections below.

"The cache reuse behavior can be controlled and the pipeline author can specify the maximum staleness of the cached data considered for reuse."

>With step caching, the output of a certain step can be re-used and we can save the efforts of re-executing previous steps.

"With execution caching enabled, the system can skip a step that has already been executed which saves time and money."